R/cmap_geom_defaults.R

```diff
@@ -0,0 +1,168 @@
+#' Initialize CMAP `default_aes` values
+#'
+#' Internal function to load in default aesthetics for modified geoms.
+#' Loaded in to `cmapplot_globals` in `.onLoad`.
+#'
+#' This list's names MUST equal `cmapplot_globals$geoms_that_change`
+#'
+#' @noRd
+init_cmap_default_aes <- function () {
+  list(
+    Line = list(
+      size = ggplot_size_conversion(cmapplot_globals$consts$lwd_plotline)
+    ),
+    Text = list(
+      family = cmapplot_globals$font$strong$family,
+      fontface = ifelse(cmapplot_globals$font$strong$face == "bold", 2, 1),
+      size = cmapplot_globals$fsize$M/ggplot2::.pt, # pt-to-mm conversion
+      colour = cmapplot_globals$colors$blackish
+    ),
+    TextLast =list(
+      family = cmapplot_globals$font$strong$family,
+      fontface = ifelse(cmapplot_globals$font$strong$face == "bold", 2, 1),
+      size = cmapplot_globals$fsize$M/ggplot2::.pt, # pt-to-mm conversion
+      colour = cmapplot_globals$colors$blackish
+    ),
+    RecessionsText = list(
+      family = cmapplot_globals$font$regular$family,
+      fontface = ifelse(cmapplot_globals$font$regular$face == "bold", 2, 1),
+      size = cmapplot_globals$fsize$S/ggplot2::.pt, # pt-to-mm conversion
+      colour = cmapplot_globals$colors$blackish
+    )
+  )
+}
+
+#' Fetch and set aesthetic defaults
+#'
+#' These functions allow for setting and resetting default aesthetic values for
+#' certain ggplot2 geoms. This is necessary for geoms to be "themed" to CMAP
+#' style standards, because (at least at the moment) setting geom aesthetic
+#' defaults on a plot-by-plot basis (such as with \code{ggplot2::theme}) is not
+#' possible. The geoms impacted are stored in
+#' \code{cmapplot_globals$geoms_to_change}.
+#'
+#' These functions are employed implictly within \code{\link{finalize_plot}} to
+#' apply preferred aesthetic defaults to final outputs. They are only necessary
+#' to use explicitly if you would like plots to use these defaults
+#' pre-\code{finalize}.
+#'
+#' CAUTION: Running \code{apply_cmap_default_aes} will set defaults for all
+#' ggplot2 plots drawn in the current session. To reset to \code{ggplot2}
+#' defaults (technically, to whatever the defaults were when \code{cmapplot}
+#' was loaded), use \code{unapply_cmap_default_aes}.
+#'
+#' Note: CMAP aesthetic defaults are loaded into
+#' \code{cmapplot_globals$default_aes_cmap} by the internal pkg function
+#' \code{init_cmap_default_aes} when \code{cmapplot} is first loaded into R.
+#'
+#' @param quietly Should the function suppress confirmation message?
+#'
+#' @examples
+#' \dontrun{
+#' g <- ggplot(mtcars, aes(x = mpg, y = disp, label = rownames(mtcars))) +
+#'   geom_line() +
+#'   geom_text() +
```

Comment:
This particular datasets produces an extremely chaotic graph, which I don't think provides a great example. I think one of the cmapplot time series datasets should be used instead.